eventsource-client/src/retry.rs

```diff
@@ -0,0 +1,155 @@
+use std::time::{Duration, Instant};
+
+pub(crate) trait RetryStrategy {
+    /// Return the next amount of time a failed request should delay before re-attempting.
+    fn next_delay(&mut self, current_time: Instant) -> Duration;
+
+    /// Modify the strategy's default base delay.
+    fn change_base_delay(&mut self, base_delay: Duration);
+
+    /// Used to indicate to the strategy that it can reset as a successful connection has been made.
+    fn reset(&mut self, current_time: Instant);
+}
+
+const DEFAULT_RESET_RETRY_INTERVAL: u64 = 60;
+
+pub(crate) struct BackoffRetry {
+    base_delay: Duration,
+    max_delay: Duration,
+    backoff_factor: u32,
+
+    reset_interval: Duration,
+    next_delay: Duration,
+    good_since: Option<Instant>,
+}
+
+impl BackoffRetry {
+    pub fn new(base_delay: Duration, max_delay: Duration, backoff_factor: u32) -> Self {
+        Self {
+            base_delay,
+            max_delay,
+            backoff_factor,
+            reset_interval: Duration::from_secs(DEFAULT_RESET_RETRY_INTERVAL),
+            next_delay: base_delay,
+            good_since: None,
+        }
+    }
+}
+
+impl RetryStrategy for BackoffRetry {
+    fn next_delay(&mut self, current_time: Instant) -> Duration {
+        let mut next_delay = self.next_delay;
+
+        if let Some(good_since) = self.good_since {
+            if current_time - good_since >= self.reset_interval {
+                self.reset(current_time);
```

Comment:
Instead of resetting it should not do that, and instead non-conditionally set the good_since to None.
(Commenting to keep track, in a huddle)

Reply:
Good catch. Resolved!